ateam_ai/src/plays/stop.hpp

```diff
@@ -0,0 +1,111 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#ifndef PLAYS__STOP_HPP_
+#define PLAYS__STOP_HPP_
+
+#include <Eigen/Dense>
+
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/point_generators_2.h>
+
+#include <vector>
+#include <cmath>
+
+#include "util/directed_graph.hpp"
+#include "types/behavior_goal.hpp"
+#include "types/field.hpp"
+#include "types/world.hpp"
+
+#include "ateam_geometry/ateam_geometry.hpp"
+
+#include "defense.hpp"
+#include "play_helpers.hpp"
+
+DirectedGraph<BehaviorGoal> generate_stop(
+  const World & world, const Field & field,
+  const FieldSidedInfo & our_side_info)
+{
+  DirectedGraph<BehaviorGoal> stop;
+
+  auto goalie = get_goalie_behavior_goal(our_side_info);
+  stop.add_node(goalie);
+
+  auto attacker_behaviors = generate_points_around_ball(world, field, 3);
+  for (BehaviorGoal behavior : attacker_behaviors) {
+    stop.add_node(behavior);
+  }
+
+  auto defense_behaviors = get_defense_behavior_goals(world, field, 2);
+  for (BehaviorGoal behavior : defense_behaviors) {
+    stop.add_node(behavior);
+  }
+  return stop;
+}
+
+std::vector<BehaviorGoal> generate_points_around_ball(
+  const World & world, const Field & field,
+  const int num_points)
+{
+  std::vector<BehaviorGoal> points_around_ball;
+  double robotDiameter = 0.18;
+  // Where is the ball?
+  std::optional<Eigen::Vector2d> ball_location = world.get_unique_ball();
+  while (!ball_location.has_value()) {
+    ball_location = world.get_unique_ball();
+  }
+  ateam_geometry::Point ball = ateam_geometry::EigenToPoint(ball_location);
+  std::vector<BehaviorGoal> points_around_ball;
+  // Used to create points around a circle with radius 0.5 m
+  Random_points_on_circle_2<ateam_geometry::Point,
+    ateam_geometry::PointCreator> circlePointGenerator;
+  circlePointGenerator circlePoints(0.55);
+  // We need 3 robots (attackers) to get close to the ball
+  while (points_around_ball.length() < 3) {
+    std::vector<ateam_geometry::Point> candidate_points;
+    candidate_points.reserve(50);
+    std::copy_n(circlePoints, 50, std::back_inserter(candidate_points));
+    // Pick the first points that are within bounds on the circle
+    // and a robot's diameter away from each other
+    ateam_geometry::Point previous_point = ateam_geometry::Point(-100, -100);
+    for (ateam_geometry::Point candidate : candidate_points) {
+      if (is_point_in_bounds(candidate)) {
+        if (ateam_geometry::Segment(
+            previous_point,
```

Comment:
Previous point is never updated with the previous point, it's always -100, -100 I think

Reply:
Added statement to update this in the loop